apph-back/src/test/java/com/viseo/apph/AuthTest.java

```diff
@@ -83,7 +83,7 @@ public void testFailLoginUserNotFind() {
         UserRequest userRequest = new UserRequest().setLogin("tintin").setPassword("password");
         when(em.createQuery("SELECT u FROM User u WHERE u.login=:login", User.class)).thenReturn(typedQuery);
```

Comment:
```suggestion
        when(em.createQuery(anyString(), eq(User.class))).thenReturn(typedQuery);
```
J'avais testé ici avec anyString, il marche bien. Dans les tests, il est mieux d'éviter des variables très précisés.

Reply:
je suis d'avis à garder les string de façon précis pour être sûre que la bonne fonctionne soit bien appelé est pas une autre createQuery avec une autre string

Reply:
Oui je comprend on pourait mettre du (anyString(), any()) aussi mais je trouve ça bien de vérifier qu'on appelle createQuery avec les bon paramètres quand même. c'est un Querry donc c'est un string mais ça correspond a des appels a une librairie donc vérifier ce string c'est comme vérifier que des bonnes fonction sont appelé avec le bon paramètre.
ça nous demande juste un peut plus de travail a chaque fois qu'on va refacto les dao. ON peut vaire un sondage pour demander a tout le monde.

Reply:
S'il y a un deuxième appel, oui c'est mieux de préciser. Mais s'il y a tout seul, je pense que c'est mieux comme ça. Parce que dans le futur, si on ajoute d'autre fonctionnalité, ça va changer le comportement d'EntityManager, donc, le test va échouer, et il informe ces changements n'est pas prévus dans les tests initiaux. L'enrichissement ne doit pas être fait au début.

Reply:
Justement si on précise le string le test ne vas pas casser si entityManager a appelé autrement a un autre endroit. Alors que si le mock est plus générique les test vas casser si em est appelé autre part avec un string différant et qu'un autre résultat est attendu

Reply:
Yep, c'est pour deux ou plusieurs fois, forcement oui. Mais si t'as un seul appel, il n'y pas de souci. Quand il échoue, donc, il signifie que t'as un autre appel dans d'autre endroit.

Reply:
Si on veut vérifier combien de fois une fonction est called on peut assert dans le THEN

Reply:
Je conserve m'avis